RFC-0034-batch-kronecker-product.md

```diff
@@ -0,0 +1,170 @@
+- Submit a pull request titled `RFC-00xx-my-feature`. 
+    - Assign the `draft` label while composing the RFC. You may find it easier to use a WYSIWYG editor (like Google Docs) when working with a few close collaborators; feel free to use whatever platform you like. Ideally this document is publicly visible and is linked to from the PR.
+    - When opening the RFC for general discussion, copy your document into the `RFC-00xx-my-feature.md` file on the PR and assign the `commenting` label.
+- Build consensus for your proposal, integrate feedback and revise it as needed, and summarize the outcome of the discussion via a [resolution template](https://github.com/pytorch/rfcs/blob/rfc-process/RFC-0000-template.md#resolution).
+    - If the RFC is idle here (no activity for 2 weeks), assign the label `stalled` to the PR.
+- Once the discussion has settled, assign a new label based on the level of support:
+    - `accepted` if a decision has been made in the RFC
+    - `draft` if the author needs to rework the RFC’s proposal
+    - `shelved` if there are no plans to move ahead with the current RFC’s proposal. We want neither to think about evaluating the proposal
+nor about implementing the described feature until some time in the future.
+- A state of `accepted` means that the core team has agreed in principle to the proposal, and it is ready for implementation. 
+- The author (or any interested developer) should next open a tracking issue on Github corresponding to the RFC.
+    - This tracking issue should contain the implementation next steps. Link to this tracking issue on the RFC (in the Resolution > Next Steps section)
+- Once all relevant PRs are merged, the RFC’s status label can be finally updated to `closed`.
+
+</details>
+
+
+# Batch Kronecker Product
+
+**Authors:**
+* @dannygoodacre
+
+
+## **Summary**
+Expand the PyTorch Kronecker product to allow for batching the product along a specified dimension.
+
+
+## **Motivation**
+
+This feature is motivated by a need for such a function in a project I am working on, which simulates the evolution of quantum spin systems. Since this feature can be generalised from matrices to tensors --- as the Kronecker product has already been in PyTorch --- I feel that it belongs within PyTorch itself.
+
+
+## **Proposed Implementation**
+In the following, `a` and `b` are tensors of the specified sizes, and `batch_kron(a, b, batch_dim)` the proposed feature, where `batch_dim = 0` refers to the outermost dimension.
+
+### In use
+For a given dimension along which to batch, the requirement is that all dimensions agree up to and including the batch dimension. For example,
+```
+a.shape = (2,3,4)
+b.shape = (2,3,5)
+```
+could be batched along dimension 0 and 1, whereas
+```
+a.shape = (3,7,2)
+b.shape = (3,5,3)
+```
+could only be batched along dimension 0.
+
+Some more examples:
+
+```
+a.shape = (2,5,4)
+b.shape = (2,5,7)
+batch_kron(a, b, 0).shape = (2,25,28)
```

Comment:
cc @zou3519 do you think that using vmap here would be the best way to expose this kind of feature rather than adding a new function? Can we implement a fast batched kron without having a native op for it?